ttnn/cpp/ttnn/tensor/types.hpp

```diff
@@ -681,6 +684,20 @@ struct Shape {
         return Shape{tt::tt_metal::Shape{this->value, tt::tt_metal::Padding{this->value.rank()}}};
     }
 
+    bool has_tile_padding() const {
+        auto rank = this->rank();
```

Comment:
you can simplify a little bit this code to something like this:
bool has_tile_padding() const {
    return std::any_of(
        std::begin(*this), 
        std::begin(*this) + this->rank(), 
        [this](int index) { return this->has_tile_padding(index); }
    );
}